src/HotChocolate/Core/src/Types/Types/Relay/GlobalIdInputValueFormatter.cs

```diff
@@ -96,7 +102,16 @@ internal class GlobalIdInputValueFormatter : IInputValueFormatter
             {
                 if (!_validateType || _typeName.Equals(idv.TypeName))
                 {
-                    list.Add(idv.Value);
+                    if (_sourceType is not null &&
```

Comment:
Source Type will never really be null. So we will always convert at this point. TryConvert is the method
https://github.com/ChilliCream/hotchocolate/blob/main/src/HotChocolate/Core/src/Types/Utilities/TypeConverterExtensions.cs#L12

which internally always calls the following method with `typeof(object)` as the first parameter:
https://github.com/ChilliCream/hotchocolate/blob/main/src/HotChocolate/Core/src/Types/Utilities/DefaultTypeConverter.cs#L43

The problem is that when the first type is `object` we reflect here the type of the value:
https://github.com/ChilliCream/hotchocolate/blob/main/src/HotChocolate/Core/src/Types/Utilities/DefaultTypeConverter.cs#L69-L71

And in case of a list we would do this for each element. 

I like the simplicity of just registering the type conversion and "it just works" though i think we could reduce the cost if we'd know the actually encoded type in the id. 
Not sure if there is a valid approach to have both, easiy configuration and "just works™️" 

There are basically two options how this could be done. 
1. When we create the serializer we check if the type is not one of the know types (meaning guid, short, int, long or string). In this case we know that we have to translate the type after serialization. on CreateSerializer we could lookup the type to translate, pass it in and then we can use the more efficent method for converting. Big plus is also that we do not leak information into the ID.

2. We extend the IdSerializer and let people to extend it more easily. The deserialization target would be encoded in the ID and we would not have to "convert" the value but more just deserialize it. The benefits of this solution are that essentially you'd be able to use more effizent ways of serializing certain types of ids as we would not have to go through the type converters.

Not sure which one of the two is the best. I am also not sure if it even is a good idea to bleed information into the serialized it (2.). From the simplicity point of view, 1. is definietly better and easier to use for users.

Definietly want to here what @michaelstaib  thinks about this. As after all the Id Serializer is hit a lot in a relay schema.